File: JavaIntroNeuralNetworkEdition2/src/com/heatonresearch/httprecipes/spider/filter/SpiderFilter.java
package com.heatonresearch.httprecipes.spider.filter;

import java.io.*;
import java.net.*;

/**
 * The Heaton Research Spider 
 * Copyright 2007 by Heaton Research, Inc.
 * 
 * HTTP Programming Recipes for Java ISBN: 0-9773206-6-9
 * http://www.heatonresearch.com/articles/series/16/
 * 
 * SpiderFilter: Filters will cause the spider to skip
 * URL's.
 * 
 * This software is copyrighted. You may use it in programs
 * This class is released under the:
 * GNU Lesser General Public License (LGPL)
 * http://www.gnu.org/copyleft/lesser.html
 * 
 * @author Jeff Heaton
 * @version 1.1
 */
public interface SpiderFilter {
  /**
   * Check to see if the specified URL is to be excluded.
   * 
   * @param url
   *          The URL to be checked.
   * @return Returns true if the URL should be excluded.
   */
  public boolean isExcluded(URL url);

  /**
   * Called when a new host is to be processed. Hosts 
   * are processed one at a time.  SpiderFilter classes 
   * can not be shared among hosts.
   * 
   * @param host
   *          The new host.
   * @param userAgent
   *          The user agent being used by the spider. Leave
   *          null for default.
   * @throws IOException
   *           Thrown if an I/O error occurs.
   */
  public void newHost(String host, String userAgent) throws IOException;
}

File: CSharpHTTPRecipesEdition1/HTTPRecipes/HeatonResearchSpider/Spider/Filter/SpiderFilter.cs
// The Heaton Research Spider for .Net Copyright 2007 by Heaton
// Research, Inc.
// 
// From the book:
// 
// HTTP Recipes for C# Bots, ISBN: 0-9773206-7-7
// http://www.heatonresearch.com/articles/series/20/
// 
// This class is released under the:
// GNU Lesser General Public License (LGPL)
// http://www.gnu.org/copyleft/lesser.html
//
using System;
using System.Collections.Generic;
using System.Text;
using System.Net;

namespace HeatonResearch.Spider.Filter
{
    /// <summary>
    /// SpiderFilter: Filters will cause the spider to skip
    /// URL's.
    /// </summary>
    public interface SpiderFilter
    {
        /// <summary>
        /// Check to see if the specified URL is to be excluded.
        /// </summary>
        /// <param name="url">The URL to be checked.</param>
        /// <returns>Returns true if the URL should be excluded.</returns>
        bool IsExcluded(Uri url);

        /// <summary>
        /// Called when a new host is to be processed. SpiderFilter
        /// classes can not be shared among hosts.
        /// </summary>
        /// <param name="host">The new host.</param>
        /// <param name="userAgent">The user agent being used by the spider. Leave
        /// null for default.</param>
        void NewHost(String host, String userAgent);
    }
}
